### PR TITLE
Add Cisco vIOS support

### PR DESCRIPTION
**Pull request by jaro0149**
- Motivation: low memory footprint compared to IOS XE/XR.
- More information is in the README.md.

### CODE REVIEW

File: vios/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:20.04
```
Comment (hellt) [inline vios/docker/Dockerfile:1]: thanks @jaro0149 
would you know if using a more recent image would work with vios?
we tend to use debian:12. you can check SR OS image for reference

File: vios/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+LABEL org.opencontainers.image.authors="xtothj@gmail.com"
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update -qy \
+   && apt-get upgrade -qy \
+   && apt-get install -y \
```
Comment (hellt) [inline vios/docker/Dockerfile:8]: we also use `--no-install-recommends` for apt to make the installation image smaller

Comment (jaro0149) [reply vios/docker/Dockerfile]: done

File: vios/docker/Dockerfile (hunk 1 above)
Comment (jaro0149) [reply vios/docker/Dockerfile]: thank you for code review

yes, it works - I have updated Dockerfile definition, I had to additionally install qemu-utils to make it work
appending docker logs
[clab-vios-lab-xrv1.log](https://github.com/user-attachments/files/17175714/clab-vios-lab-xrv1.log)

**Comment by hellt** [comment]
Thanks @jaro0149 
I pushed a little change to have a simple version pattern (hate sed for complex regexps). The rest I haven't touched.

@kaelemc is it something you would be interested in trying before the merge?

**Comment by kaelemc** [comment]
@hellt tested and it works great, did a basic test between two nodes and got an OSPF adjacency up.

By default the images in CML are formatted as `vios-adventerprisek9-m.spa.159-3.m3.qcow2`. Using the change to the version matching it means users have to rename the image. If we used the one that @jaro0149 originally committed I would make one change to match on `[spa|SPA]` as some images have `spa` in lowercase and some in uppercase.

I think it should also be it's own kind in containerlab instead of using the `linux` kind. That way we can also get the nice interface aliases instead of using `ethX`.

Nice work @jaro0149.

**Comment by hellt** [comment]
I deliberately changed the version to be man-made. It was often a problem to match the versions when a vendor changes its naming schema. It is all easier to ask a human to rename the qcow to a well known pattern, rather than expect them to get an image that we can parse.

Thanks for your review, I agree it might make sense to have a kind in clab to make the interface aliasing to work. This though can be merged it seems, and I will proceed with merging.

@jaro0149 thanks for your contribution!

**Pull request merged**